src/persist/src/indexed/encoding.rs

```diff
@@ -330,8 +340,26 @@ impl BlobTraceMeta {
     /// Asserts Self's documented invariants, returning an error if any are
     /// violated.
     pub fn validate(&self) -> Result<(), Error> {
+        let upper = self.ts_upper();
+        let min = Antichain::from_elem(Timestamp::minimum());
+
+        if self.since != min && !PartialOrder::less_than(&self.since, &upper) {
```

Comment:
yeah exactly -- the invariant is that since is either 0 or less than upper (seemed more compact than making it dependent on `if upper == 0`. can restructure the invariant if that helps

Reply:
nope, this is fine. just checking my understanding